### PR TITLE
fix(nuxt3): render nuxt custom error page

### LINKED ISSUE #4250: error page can't custom render, error page rendered by nitropack
### Environment

Nuxt CLI v3.0.0-27491748.3186841                                        17:25:24
RootDir: /Users/liuyongjian/kezhi/k1022/nuxt3-app                       17:25:26
Nuxt project info:                                                      17:25:26

------------------------------
- Operating System: `Darwin`
- Node Version:     `v16.14.2`
- Nuxt Version:     `3.0.0-27491748.3186841`
- Package Manager:  `yarn@1.22.18`
- Builder:          `vite`
- User Config:      `publicRuntimeConfig`, `css`, `meta`, `vite`
- Runtime Modules:  `-`
- Build Modules:    `-`
------------------------------

👉 Report an issue: https://github.com/nuxt/framework/issues/new        17:25:26

👉 Suggest an improvement: https://github.com/nuxt/framework/discussions/new

👉 Read documentation: https://v3.nuxtjs.org

### Reproduction

;

### Describe the bug

I create error.vue, updated lastest nuxt3. the lastest nuxt3 used nitropack, it can't render custom error page

### Additional context

_No response_

### Logs

_No response_

### PR DESCRIPTION
**Pull request by danielroe**
### 🔗 Linked issue

resolves  #4250
 
### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Follow up on https://github.com/nuxt/framework/issues/4250#issuecomment-1094844435.

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by danielroe** [comment]
@pi0 Given we use new defu in nitropack, we may have to bump bridge defu as well..

**Comment by pi0** [comment]
> Given we use new defu in nitropack, we may have to bump bridge defu as well, but IIRC we had some breaking issues last time. WDYT?

Updating is fine as long as using default export instead of named one (until nuxt 2 also bumps defu. until then hosting can make issues)

**Comment by danielroe** [comment]
**Note**: we should add a breaking change indicator for bridge and require people to add the new version of `defu` to their `package.json` (as we now require this in nitropack 0.2.4).

File: test/fixtures/bridge/package.json [hunk 1]
```diff
@@ -7,6 +7,7 @@
   },
   "devDependencies": {
     "@nuxt/bridge": "*",
+    "defu": "^6.0.0",
```
Comment (pi0) [inline test/fixtures/bridge/package.json:10]: This can be also a direct dependcy. Issue is not with newer version but old version being hoisted to top!

Comment (danielroe) [reply test/fixtures/bridge/package.json:10]: Yes, that's right. v6 is already a direct dependency of `@nuxt/bridge`, and adding this directly in user's project is likely the best way to ensure top-level defu is v6.

**Comment by pi0** [comment]
> Note: we should add a breaking change indicator for bridge and require people to add the new version of defu to their package.json (as we now require this in nitropack 0.2.4) - or we need to revert this in nitropack.

I'm not sure about default hosting behavior with published package. Let's try on edge and see if there is a better way to enforce bridge using latest defu.

File: packages/bridge/src/app.ts [hunk 2]
```diff
@@ -82,17 +82,9 @@ export function setupAppBridge (_options: any) {
...
-  const defuPath = genString(_require.resolve('defu'))
-  nuxt.options.alias.defu = addTemplate({
-    filename: 'defu.alias.mjs',
-    getContents: () => [
-      `import { defu } from ${defuPath};`,
-      `export * from ${defuPath};`,
-      'export { defu as default };'
-    ].join('\n')
-  }).dst
+  nuxt.options.alias.defu = _require.resolve('defu')
```
Comment (pi0) [inline packages/bridge/src/app.ts:87]: This can be reverted with latest nitro.

File: packages/kit/src/module/define.ts [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import { promises as fsp } from 'fs'
-import defu from 'defu'
+import { defu } from 'defu'
```
Comment (pi0) [inline packages/kit/src/module/define.ts:2]: This should be reverted too from this PR.

**Pull request merged**